### ISSUE TITLE
Smart Quote Error (Warning?)

### ISSUE THREAD

**Issue by stephanecopin** [danger/danger#977]
# Report

## What did you do?

I've ran `danger local`, but this is reproducable on all invocation of Danger.

## What did you expect to happen?
No errors should have been reported regarding smart quotes.

## What happened instead?
The following error was reported:
```
Your Dangerfile has had smart quotes sanitised. To avoid issues in the future, you should not use TextEdit for editing it. If you are not using TextEdit, you should turn off smart quotes in your editor of choice.
```

## Your Environment

* Which CI are you running on? Bitrise, but unrelevant in this case
* Are you running the latest version of Danger? Yes, 5.5.11
* What is your Dangerfile?

```ruby
 warn 'This “should“ reproduce the warning'
```

## Notes

While the `Dangerfile` that reproduces the error (warning?) is trivial, what I'm currently experiencing is that one of the extensions is reporting a message with smart quotes instead. While the extension we're using is custom, I don't see any reasons for it not to happen for others.
It's probably a low priority issue, considering that it's non-breaking most of the times, but technically valid Dangerfile such as these:

```ruby
 error "This “should“ output an error"
```

then fails to be parsed correctly.

**Comment by orta** [comment]
This is pretty edge-casey, you're welcome to take a look at it, but I'd rather the warning be there for non-ruby programmers to understand when we've auto-fixed something for them.

**Comment by orta** [comment]
A fix for most cases could be to find the line with smart quotes, strip it of whitespace and see if it starts with `warn`, `message`, `fail` etc

**Comment by stephanecopin** [comment]
@orta I agree, that'd be a good fix for probably most of the times it'd be an issue. I'll try to look into it this weekend

**Comment by danielgomezrico** [comment]
It would be awesome if the tool could print on which line the smarts quotes can be found, I am not been able to find where are they 🙈
